wordpress-mobile/WordPress-iOS
# TimeZone selection screen - add contextual labels for GMT Offset timezone, CurrentTime

Timezone selection from site settings in iOS app is outdated in comparison with WordPress web and Android. This PR brings iOS on par with them by adding extra labels into the TableView to show for each WPTimeZone
 * Relative GMT Offset 
 * Current time of each WPTimeZone

Ref #11450

Android [Issue](https://github.com/wordpress-mobile/WordPress-Android/issues/7900) and [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/14198) for reference


### To test:

To test this UI enhancement:

1. Navigate to My Site -> Site Settings -> Time Zone
2. Make sure the time zone that is suggested is consistent with the timezone of the device / simulator you are testing on
3. Check that there's a GMT offset string below the TimeZone primary title and the local time for that TimeZone is shown as a subtitle on the right hand side

### Screenshot of enhancement
![image](https://user-images.githubusercontent.com/88816724/134630733-db1356cd-7e21-4c2a-8f86-e24938666bdd.png)


Additionally I've tested the following with accompanying screenshots:
 * Issue task item "Ensure view(s) are correct on iPhone and iPad" - Portrait and Landscape screenshots
 
![image](https://user-images.githubusercontent.com/88816724/134631170-5a29b2e8-68c6-4c5a-b5b4-4b4e5b176efa.png)

 
 * Issue task item "Ensure view(s) are correct in both light and dark appearance" - dark appearance screenshot
![image](https://user-images.githubusercontent.com/88816724/134631449-fd2b2f03-ffd5-49e5-8e1f-c29616489f64.png)


 * Issue task item "Add accessibility support if needed".  2 images:
   * Accessibility with large text and 
   * Accessibility Hint for button
![image](https://user-images.githubusercontent.com/88816724/134630949-0ec7a9f3-cba3-4237-bc6a-b04343df9112.png)


![image](https://user-images.githubusercontent.com/88816724/134631024-13fa6fe1-326d-44e3-8f88-e88567be3941.png)




## Regression Notes
1. Potential unintended areas of impact
 * TimeZone selector screen was changed - unintended impact could be user would not be able to change their TimeZone
 * TimeZoneSuggestion feature (PR #17145) could no longer work

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What I did to test those areas of impact (or what existing automated tests I relied on)
 * I retested all steps in (PR #17145)
   * Navigate to My Site -> Site Settings -> Time Zone
   * Make sure the time zone that is suggested is consistent with the timezone of the device / simulator you are testing on
   * Tap the suggested timezone
   * This will navigate you back to Site Settings screen where you should see the Suggested time you have chosen
   * Tap Time Zone again from Site Settings screen
   * To test search results works normally and has no regression:
   * Search for a timezone (example Casablanca). TableView results should filter and show Casablanca.
   * Tap Casablanca
   * This will navigate you back to Site Settings screen where you should see the Suggested time you have chosen (Casablanca)



3. What automated tests I added (or what prevented me from doing so)
No automated tests existed previously and ViewController is not in a state that facilitates automated testing

PR submission checklist:
